File: experiments/complexity_analysis/O1_constant/slow_solutions/slow_solution.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    long long a, b;
    cin >> a >> b;
    
    // Algorithmically equivalent but inefficient implementation
    // Using inefficient methods for basic arithmetic operations
    
    // Inefficient sum: using repeated addition
    long long sum = a;
    if (b > 0) {
        for (long long i = 0; i < b; i++) sum++;
    } else if (b < 0) {
        for (long long i = 0; i < -b; i++) sum--;
    }
    cout << sum << endl;
    
    // Inefficient difference: using repeated subtraction  
    long long diff = a;
    if (b > 0) {
        for (long long i = 0; i < b; i++) diff--;
    } else if (b < 0) {
        for (long long i = 0; i < -b; i++) diff++;
    }
    cout << diff << endl;
    
    // Inefficient product: using repeated addition
    long long product = 0;
    long long abs_b = (b >= 0) ? b : -b;
    for (long long i = 0; i < abs_b; i++) {
        product += a;
    }
    if (b < 0) product = -product;
    cout << product << endl;
    
    // Integer division (handle division by zero)
    if (b != 0) {
        cout << a / b << endl;
    } else {
        cout << 0 << endl;
    }
    
    return 0;
}

File: experiments/complexity_analysis/O1_constant/slow_solutions/slow_solution.py
a, b = map(int, input().split())

# Algorithmically equivalent but inefficient implementation
# Using inefficient methods for basic arithmetic operations

# Inefficient sum: using repeated addition
sum_result = a
if b > 0:
    for i in range(b):
        sum_result += 1
elif b < 0:
    for i in range(-b):
        sum_result -= 1
print(sum_result)

# Inefficient difference: using repeated subtraction
diff_result = a
if b > 0:
    for i in range(b):
        diff_result -= 1
elif b < 0:
    for i in range(-b):
        diff_result += 1
print(diff_result)

# Inefficient product: using repeated addition
product_result = 0
abs_b = abs(b)
for i in range(abs_b):
    product_result += a
if b < 0:
    product_result = -product_result
print(product_result)

# Integer division (handle division by zero)
if b != 0:
    print(a // b)
else:
    print(0)
